test/fluree/db/query/fql_test.clj

```diff
@@ -238,6 +238,30 @@
           (is (= [["ex:bob"]] sut)
               "returns correctly filtered results"))))))
 
+(deftest ^:integration ^:pending datatype-test
+  (let [conn   (test-utils/create-conn)
+        ledger @(fluree/create conn "people"
+                               {:defaultContext
+                                ["" {:ex "http://example.org/ns/"}]})
+        db     @(-> ledger
+                    fluree/db
+                    (fluree/stage
+                     [{:id      :ex/homer
+                       :ex/name "Homer"
+                       :ex/age  36}
+                      {:id      :ex/bart
+                       :ex/name "Bart"
+                       :ex/age  "forever 10"}]))]
+    (testing "including datatype in query results"
+      (let [query   '{:select [?age ?dt]
+                      :where  [[?s :ex/age ?age]
+                               {:bind {?dt (datatype ?age)}}]}
+            results @(fluree/query db query)]
+        (is (= [["forever 10" 1] [36 8]]
```

Comment:
When the datatype filtering is implemented, will the datatypes returned by queries be iris?

Reply:
Yeah that's the plan. I can update the test in here to expect that.